dawn/src/dawn/IIR/Extents.cpp

```diff
@@ -16,59 +16,68 @@
 #include "dawn/Support/Assert.h"
 #include "dawn/Support/StringUtil.h"
 #include "dawn/Support/Unreachable.h"
+#include <algorithm>
 #include <iostream>
 
 namespace dawn {
 namespace iir {
 
-Extents::Extents(const ast::Offsets& offset) {
+Extents::Extents(ast::cartesian_, const ast::Offsets& offset)
+    : horizontalExtent_(ast::cartesian_{}) {
   auto const& hOffset = ast::offset_cast<ast::CartesianOffset const&>(offset.horizontalOffset());
-  std::array<int, 3> offsetArray = {hOffset.offsetI(), hOffset.offsetJ(), offset.verticalOffset()};
 
-  for(int i = 0; i < extents_.size(); ++i)
-    extents_[i].Minus = extents_[i].Plus = offsetArray[i];
+  horizontalExtent_ = HorizontalExtent(ast::cartesian_{}, hOffset.offsetI(), hOffset.offsetI(),
+                                       hOffset.offsetJ(), hOffset.offsetJ());
+  verticalExtent_ = Extent(offset.verticalOffset(), offset.verticalOffset());
 }
 
-Extents::Extents(int extent1Minus, int extent1Plus, int extent2Minus, int extent2Plus,
-                 int extent3Minus, int extent3Plus)
-    : extents_({{{extent1Minus, extent1Plus},
-                 {extent2Minus, extent2Plus},
-                 {extent3Minus, extent3Plus}}}) {}
+Extents::Extents(ast::cartesian_, int extent1minus, int extent1plus, int extent2minus,
+                 int extent2plus, int extent3minus, int extent3plus)
+    : horizontalExtent_(ast::cartesian_{}) {
+  horizontalExtent_ =
+      HorizontalExtent(ast::cartesian_{}, extent1minus, extent1plus, extent2minus, extent2plus);
+  verticalExtent_ = Extent(extent3minus, extent3plus);
+}
+
+Extents::Extents(ast::cartesian_) : horizontalExtent_(ast::cartesian_{}) {}
+
+Extents::Extents(const Extents& other)
+    : verticalExtent_(other.verticalExtent()), horizontalExtent_(other.horizontalExtent()) {}
```

Comment:
alternative:
Extents::Extents(Extents const& other) {
    *this = other;
}
Extents::Extents(Extents&& other) {
    *this = std::move(other);
}